### PR TITLE
Trigger rebuild for opinion changes

### PR DESCRIPTION
**Pull request by andreas-kupries**
Trello: https://trello.com/c/IemiPFAX/105-fissile-doesn-t-detect-opinion-changes-so-it-doesn-t-rebuild-images

Stacked on https://github.com/SUSE/fissile/pull/239
because of the similarity (same area of code),  tested together.

### CODE REVIEW

File: model/roles.go [hunk 1]
```diff
@@ -854,3 +854,39 @@ func (r *Role) IsDevRole() bool {
 	}
 	return false
 }
+
+// GetFileSignature returns the SHA1 of a file based on its path and contents
```
Comment (alexjh) [inline model/roles.go]: I don't see this function being used anywhere.

File: model/roles.go [hunk 2]
```diff
@@ -854,3 +854,39 @@ func (r *Role) IsDevRole() bool {
...
+func GetFileSignature(filePath string) (string, error) {
```
Comment (alexjh) [inline model/roles.go]: If this function is being used, I think we should make it a common utility function and share it between `model/job.go`, `model/package.go`, and this file. Looks like there are similar things being done in all of them.

File: model/roles.go (hunk 1 above)
Comment (andreas-kupries) [inline model/roles.go]: Right. I used used in earlier revisions of the code and then it fell by the wayside.
Will remove.

File: model/roles.go (hunk 2 above)
Comment (andreas-kupries) [inline model/roles.go]: Mooted

File: model/roles.go [hunk 3]
```diff
@@ -854,3 +854,39 @@ func (r *Role) IsDevRole() bool {
...
+	hasher := sha1.New()
+
+	hasher.Write([]byte(filePath))
+
+	f, err := os.Open(filePath)
+	if err != nil {
+		return "", err
+	}
+
```
Comment (andreas-kupries) [inline model/roles.go]: Mooted

File: model/roles.go (hunk 1 above)
Comment (andreas-kupries) [inline model/roles.go]: Done with commit c1b5d3cb6eec7bd263ac9033b7a8efc8116b11eb.

File: model/roles.go (hunk 2 above)
Comment (andreas-kupries) [inline model/roles.go]: Done with commit c1b5d3cb6eec7bd263ac9033b7a8efc8116b11eb.

File: model/roles.go (hunk 3 above)
Comment (andreas-kupries) [inline model/roles.go]: Done with commit c1b5d3cb6eec7bd263ac9033b7a8efc8116b11eb.

**Pull request merged**